File: FDFrontend/src/app/model/restaurant.ts
import { Administrator } from "./administrator";
import { DeliveryZone } from "./deliveryZone";
import { Menu } from "./menu";

export class Restaurant{
    public id: number | null;
    public name: string | null;
    public location: string | null;
    public administrator: Administrator;
    public menus: Menu[] | null;
    public deliveryZones: DeliveryZone[] | null;

    constructor(){
        this.id = null;
        this.name = null;
        this.location = null;
        this.menus = [];
        this.deliveryZones = [];
        this.administrator = new Administrator();
    }
}

File: FDBackend/src/main/java/FDBackend/Entities/Restaurant.java
package FDBackend.Entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "restaurants")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Restaurant implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column
    private String name;

    @Column
    private String location;

    @OneToOne(cascade = CascadeType.MERGE)
    @JoinColumn(name = "administratorId",unique = true)
    private Administrator administrator;

    @OneToMany(mappedBy = "restaurant",fetch = FetchType.LAZY,cascade = CascadeType.PERSIST)
    private List<Menu> menus;

    @OneToMany(mappedBy = "restaurant",fetch = FetchType.LAZY,cascade = CascadeType.PERSIST)
    private List<DeliveryZone> deliveryZones = new ArrayList<>();
}
